cpp/src/arrow/csv/test_common.h

```diff
@@ -47,7 +47,7 @@ ARROW_TESTING_EXPORT
 void MakeColumnParser(std::vector<std::string> items, std::shared_ptr<BlockParser>* out);
 
 ARROW_TESTING_EXPORT
-Result<std::shared_ptr<Buffer>> MakeSampleCsvBuffer(size_t num_rows, bool valid = true);
+Result<std::shared_ptr<Buffer>> MakeSampleCsvBuffer(size_t num_rows, int invalid = 0);
```

Comment:
`size_t num_invalid_rows` perhaps?

Reply:
Or instead perhaps pass an optional predicate: `std::function<bool(int row_num)> is_valid`.
This will make the test values less magical.

Reply:
Changed it to a predicate function